Sources/ResilientDecoding/Resilient.swift

```diff
@@ -15,163 +15,105 @@ public struct Resilient<Value: Decodable>: Decodable {
   public init(from decoder: Decoder) throws {
     assertionFailure()
     let value = try Value(from: decoder)
-    self = Self(value)
+    self = Self(value, outcome: .decodedSuccessfully)
   }
 
   /**
    Initialized a `Resilient` value as though it had been decoded without encountering any errors.
    */
   public init(_ value: Value) {
-    wrappedValue = value
-    #if DEBUG
-    projectedValue = PropertyLevelErrors(decodedValue: value, error: nil)
-    #endif
+    self.wrappedValue = value
+    self.outcome = .decodedSuccessfully
   }
-
-  #if DEBUG
-
-  fileprivate init(_ value: Value, error: Error)
-  {
-    wrappedValue = value
-    projectedValue = PropertyLevelErrors(decodedValue: value, error: error)
+    
+  init(_ value: Value, outcome: ResilientDecodingOutcome) {
+    self.wrappedValue = value
+    self.outcome = outcome
   }
 
-  public let projectedValue: PropertyLevelErrors
-
-  #endif
-
   public let wrappedValue: Value
 
+  let outcome: ResilientDecodingOutcome
+
   /**
    Transforms the value of a `Resilient` type.
-   If `wrappedValue` is an array,  care should be taken to ensure that the `value.count` == `transform(value).count` in order to not break the `results` property.
+   If `self` is a resilient array,  care should be taken to ensure that the `value.count` == `transform(value).count` in order to not break the `results` property.
    */
   func map<T>(transform: (Value) -> T) -> Resilient<T> {
-    let value = transform(wrappedValue)
-    #if DEBUG
-    if let error = projectedValue.error {
-      return Resilient<T>(value, error: error)
-    } 
-    #endif
-    return Resilient<T>(value)
+    Resilient<T>(transform(wrappedValue), outcome: outcome)
   }
-}
-
-// MARK: - Creating Resilient Values
-
-extension Decoder {
-
-  /**
-   Creates a `Resilient` value with a fallback value after an error has occurred and reports the error to `ResilientDecodingErrorReporter`.
-   Since this is the only way to create a `Resilient` with an error, this ensures we are reporting all errors encountered in this manner.
-   */
-  func resilient<T>(_ fallbackValue: T, error: Error) -> Resilient<T> {
-    #if DEBUG
-      if error is ArrayDecodingError {
-        /// Resilient arrays are reponsible for reporting element decoding errors themselves.
-      } else {
-        resilientDecodingHandled(error)
+  
+  #if DEBUG
+  @dynamicMemberLookup
```

Comment:
Can you help me understand what this dynamic member lookup is doing? I thought we had to add a method to this type for it to have any effect, and I'm not seeing one... But I could be missing it in another file or part of this diff.

Reply:
The discussion [here](https://forums.swift.org/t/gauging-community-sentiment-optionalprotocol/35269/5?u=george) resulted in [this PR](https://github.com/airbnb/ResilientDecoding/pull/11). The definition of the subscript is in `ResilientArray+DecodingOutcome`, the only reason they are separate is because `@dynamicMemberLookup` can't be defined on an extension. I'll add a comment.

Reply:
Cool. You'll be happy to know that because of your work on this I read the entire property wrappers Swift evolution proposal in extreme detail last weekend. Trying to make sure I'm on the same page...

Reply:
To be entirely honest I've read that forum post probably 5 times over multiple days, and I can't really understand the problem. Can you give me a quick summary? Maybe that will help me approach it in a new light. I'm at a loss currently.